pandas/tests/test_categorical.py

```diff
@@ -797,6 +797,149 @@ def test_set_categories_inplace(self):
         cat.set_categories(['a', 'b', 'c', 'd'], inplace=True)
         tm.assert_index_equal(cat.categories, pd.Index(['a', 'b', 'c', 'd']))
 
+    @pytest.mark.parametrize(
+        "input1, input2, cat_array",
+        [
+            (
```

Comment:
so instead of duplicating code like this you can use multiple parametrize
just parametrize on the dtype and construct inside
something like
```
@pytest.mark.parmetrize(
"consturctor", ['int_', ....])
def test_drop_duplicates_non_bool(self, input):
      input1 = np.array([1, 2, 3, 3], dtype=dtype)
      ....
```

this will work for datetimes and timedeltas as well (use 'M8' and 'm8')